subscriber/check.go

```diff
@@ -0,0 +1,59 @@
+package subscriber
+
+import (
+	"context"
+
+	"github.com/docker/distribution/reference"
+	"github.com/grafeas/voucher"
+	"github.com/grafeas/voucher/cmd/config"
+	"github.com/grafeas/voucher/repository"
+)
+
+// check runs all checks for a given image.
+// Returns true if the required check(s) have passed and true if the check run needs to be retried.
+func (s *Subscriber) check(canonicalImageReference reference.Canonical) (bool, bool) {
+	var repositoryClient repository.Client
+	var err error
+
+	ctx, cancel := context.WithTimeout(context.Background(), s.cfg.TimeoutDuration())
+	defer cancel()
+
+	metadataClient, err := config.NewMetadataClient(ctx, s.secrets)
+	if nil != err {
+		s.log.Errorf("failed to create MetadataClient: %s", err)
+		return false, true
+	}
+	defer metadataClient.Close()
+
+	buildDetail, err := metadataClient.GetBuildDetail(ctx, canonicalImageReference)
```

Comment:
Could we move this logic into the `cmd/config` package? Specifically, line 28 through to line 40!

Reply:
i'm not sure I understand why! these take the `canonicalImageReference` as input so I think it makes most sense to leave that logic here instead of `cmd/config`?

Reply:
I think the main reason would be to simplify the logic in the two servers, mostly because this function is very long presently and that represents an understandability burden! At the very least, I'd argue we should move this code into a separate function, that way we don't have something that we expect to fail some percentage of the time taking up so 30 lines of code!

Reply:
so based off of @ooq's comment below, should we move logic to a central spot for checking? i'm thinking of creating a `checks/checker.go` file that has a `Check` function that will do all of this logic and then `server/check.go` will call out to this and so will `subscriber/check.go`. what do you think?

Reply:
So the reason I would argue for `cmd/config` is because this code is dependent on the `config.NewRepositoryClient` function, which shouldn't be imported in the rest of the code base (since we'll get import loops). We might be able to move this into `checks` after refactoring the secrets logic but for now I'd argue for simple refactors 😅

Reply:
As per our discussion, we're going to do a followup refactor! :)